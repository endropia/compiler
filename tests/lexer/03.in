and array asm begin break
case const constructor continue
destructor div do downto
else end false file for
function goto if implementation in
inline interface label mod nil
not object of on
operator or packed procedure
program record repeat set shl shr
then to true type
unit until uses var
while with xor as
class constref dispose except
exit exports finalization finally
inherited initialization is library
new on out property
raise self threadvar
try absolute abstract alias
assembler cdecl Cppdecl default export
external forward generic index
local name nostackframe oldfpccall
override pascal private protected
public published read register
reintroduce safecall softfloat specialize
stdcall virtual write string
String
STRING