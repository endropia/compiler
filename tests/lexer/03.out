1	1	Keyword	AND	and
1	5	Keyword	ARRAY	array
1	11	Keyword	ASM	asm
1	15	Keyword	BEGIN	begin
1	21	Keyword	BREAK	break
2	1	Keyword	CASE	case
2	6	Keyword	CONST	const
2	12	Keyword	CONSTRUCTOR	constructor
2	24	Keyword	CONTINUE	continue
3	1	Keyword	DESTRUCTOR	destructor
3	12	Keyword	DIV	div
3	16	Keyword	DO	do
3	19	Keyword	DOWNTO	downto
4	1	Keyword	ELSE	else
4	6	Keyword	END	end
4	10	Keyword	FALSE	false
4	16	Identifier	file	file
4	21	Keyword	FOR	for
5	1	Keyword	FUNCTION	function
5	10	Keyword	GOTO	goto
5	15	Keyword	IF	if
5	18	Keyword	IMPLEMENTATION	implementation
5	33	Keyword	IN	in
6	1	Keyword	INLINE	inline
6	8	Keyword	INTERFACE	interface
6	18	Keyword	LABEL	label
6	24	Keyword	MOD	mod
6	28	Keyword	NIL	nil
7	1	Keyword	NOT	not
7	5	Keyword	OBJECT	object
7	12	Keyword	OF	of
7	15	Keyword	ON	on
8	1	Keyword	OPERATOR	operator
8	10	Keyword	OR	or
8	13	Keyword	PACKED	packed
8	20	Keyword	PROCEDURE	procedure
9	1	Keyword	PROGRAM	program
9	9	Keyword	RECORD	record
9	16	Keyword	REPEAT	repeat
9	23	Keyword	SET	set
9	27	Keyword	SHL	shl
9	31	Keyword	SHR	shr
10	1	Keyword	THEN	then
10	6	Keyword	TO	to
10	9	Keyword	TRUE	true
10	14	Keyword	TYPE	type
11	1	Keyword	UNIT	unit
11	6	Keyword	UNTIL	until
11	12	Keyword	USES	uses
11	17	Keyword	VAR	var
12	1	Keyword	WHILE	while
12	7	Keyword	WITH	with
12	12	Keyword	XOR	xor
12	16	Keyword	AS	as
13	1	Keyword	CLASS	class
13	7	Keyword	CONSTREF	constref
13	16	Keyword	DISPOSE	dispose
13	24	Keyword	EXCEPT	except
14	1	Keyword	EXIT	exit
14	6	Keyword	EXPORTS	exports
14	14	Keyword	FINALIZATION	finalization
14	27	Keyword	FINALLY	finally
15	1	Keyword	INHERITED	inherited
15	11	Keyword	INITIALIZATION	initialization
15	26	Keyword	IS	is
15	29	Keyword	LIBRARY	library
16	1	Keyword	NEW	new
16	5	Keyword	ON	on
16	8	Keyword	OUT	out
16	12	Keyword	PROPERTY	property
17	1	Keyword	RAISE	raise
17	7	Keyword	SELF	self
17	12	Keyword	THREADVAR	threadvar
18	1	Keyword	TRY	try
18	5	Keyword	ABSOLUTE	absolute
18	14	Keyword	ABSTRACT	abstract
18	23	Keyword	ALIAS	alias
19	1	Keyword	ASSEMBLER	assembler
19	11	Keyword	CDECL	cdecl
19	17	Keyword	CPPDECL	Cppdecl
19	25	Keyword	DEFAULT	default
19	33	Keyword	EXPORT	export
20	1	Keyword	EXTERNAL	external
20	10	Keyword	FORWARD	forward
20	18	Keyword	GENERIC	generic
20	26	Keyword	INDEX	index
21	1	Keyword	LOCAL	local
21	7	Keyword	NAME	name
21	12	Keyword	NOSTACKFRAME	nostackframe
21	25	Keyword	OLDFPCCALL	oldfpccall
22	1	Keyword	OVERRIDE	override
22	10	Keyword	PASCAL	pascal
22	17	Keyword	PRIVATE	private
22	25	Keyword	PROTECTED	protected
23	1	Keyword	PUBLIC	public
23	8	Keyword	PUBLISHED	published
23	18	Keyword	READ	read
23	23	Keyword	REGISTER	register
24	1	Keyword	REINTRODUCE	reintroduce
24	13	Keyword	SAFECALL	safecall
24	22	Keyword	SOFTFLOAT	softfloat
24	32	Keyword	SPECIALIZE	specialize
25	1	Keyword	STDCALL	stdcall
25	9	Keyword	VIRTUAL	virtual
25	17	Keyword	WRITE	write
25	23	Keyword	STRING	string
26	1	Keyword	STRING	String
27	1	Keyword	STRING	STRING
27	7	eof
